flink-architecture-tests/src/test/resources/archunit.properties

```diff
@@ -0,0 +1,24 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Path under which known violations will be stored for frozen rules
+freeze.store.default.path=violations
+
+# Enable this to update the violations files
```

Comment:
I still want to add more documentation here, but I'm still playing around with the best setup. On the CI, changes to violations should always fail. For local execution we can decide between failing and automatically updating the violation store. The latter is more convenient, but has a slight risk that people just "accept" those changes. I'm not sure what the better approach is (yet).

Reply:
Since there are separate settings for creating and updating violation stores, my preferred approach at the moment would be to automatically update stores locally, but not create them. On CI both are disabled. I think this reduces developer frustration a bit since resolving a violation would automatically be recorded and not require changing some config and running the test again.

Reply:
Is there no way to configure it to work as remove-only? If a violation is fixed, remove entry. If a new one is added, fail.

That would be the only "sane" approach in my opinion. If by default we just add a violation then no one will fix this stuff.

Reply:
We could implement our own violation store, but in the default one that doesn't seem to be possible. I've raised https://github.com/TNG/ArchUnit/issues/670 for this for now. I don't think I'd start with a custom violation store for the time being and instead have a separate target which can be run to update violations, but disallow updates by default (local & CI).

Reply:
Based on someone's feedback I actually verified that this already behaves the way we want to, i.e. when run locally new violations will lead to a failure, but removed violations will be updated. On the CI we fail for any change to the violation store, even removal, since that removal needs to be committed.